### PR TITLE
Support generic aliases in all places where types or tuple of types are allowed

### PR DESCRIPTION
**Pull request by MatthieuDartiailh**
Tests and extra documentation are missing but I otherwise this to be reasonably complete. Feel free to review.

### CODE REVIEW

**Comment by MatthieuDartiailh** [comment]
@frmdstryr and @clamdad I believe this is finally ready for review and with it 0.7.0 would be ready I believe.

File: atom/instance.py [hunk 1]
```diff
@@ -6,8 +6,10 @@
 # The full license is in the file LICENSE, distributed with this software.
 #------------------------------------------------------------------------------
 import sys
+from itertools import chain
```
Comment (frmdstryr) [inline atom/instance.py:9]: Unused?

Comment (MatthieuDartiailh) [reply atom/instance.py:9]: Good catch !

File: atom/instance.py [hunk 2]
```diff
@@ -6,8 +6,10 @@
...
 
 from .catom import DefaultValue, Member, Validate
+from .typing_utils import extract_types, is_optional
 
 if sys.version_info >= (3, 9):
     from typing import GenericAlias
```
Comment (clamdad) [inline atom/instance.py:15]: Also unused?

**Comment by MatthieuDartiailh** [comment]
Thanks for catching those @frmdstryr and @clamdad I will merge and publish a new rc for atom (and merge https://github.com/nucleic/enaml/pull/458 and release a rc for enaml).
I will likely make the final release early next week.

**Pull request merged**